### PR TITLE
Improve the Mockup documentation

### PR DESCRIPTION
**Pull request by thet**
This is a preparation to include the Mockup documentation in the Plone documentation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -232,3 +252,78 @@ The translations are handled by ``src/core/i18n.js``.
 This translation helper that calls the ``@@plonejsi18n`` view defined in plone.app.content to generate a JSON of the translations from the mo file.
 The ``@@plonejsi18n`` view is called one time for a given domain and language and the result is cached in localStorage for 24 hours.
 The only way to test the new translations is to restart the instance to update the mo file from the po file, and then to purge the localStorage to trigger a new download of the translations.
+
+
+### Styleguide
```
Comment (stevepiercy) [inline README.md:257]:
```suggestion
## Style guide
```

File: README.md [hunk 2]
```diff
@@ -232,3 +252,78 @@ The translations are handled by ``src/core/i18n.js``.
...
+**And this is how you use it:**
+
+We have 4 different types of changelog entries:
+
+-   Breaking Changes (type: `breaking`),
+-   Features (type: `feat`),
+-   Bug Fixes (type: `fix`),
+-   Maintenance (type: `maint`).
+
+We can group commits in the changelog via a scope - or grouping.
```
Comment (stevepiercy) [inline README.md:277]:
```suggestion
We can group commits in the changelog via a scope or grouping.
```

File: README.md [hunk 3]
```diff
@@ -232,3 +252,78 @@ The translations are handled by ``src/core/i18n.js``.
...
+Let's follow a convention and use these groupings - but the grouping is optional and any other group name can be used.
```
Comment (stevepiercy) [inline README.md:278]:
```suggestion
Let's follow a convention and use these groupings, but the grouping is optional and any other group name can be used.
```

File: README.md [hunk 4]
```diff
@@ -232,3 +252,78 @@ The translations are handled by ``src/core/i18n.js``.
...
+
+-   "Dependencies" for upgrading package.json dependencies.
+-   "Docs" for documentation.
+-   "Build" for everything related to the release workflow, Webpack and building bundles.
+-   "Cleanup" for cleaning up or reformatting code.
+-   "pat PATTERNNAME" for changes to individual patterns in src/pat.
```
Comment (stevepiercy) [inline README.md:284]:
```suggestion
-   "pat PATTERNNAME" for changes to individual patterns in `src/pat`.
```

File: README.md [hunk 5]
```diff
@@ -232,3 +252,78 @@ The translations are handled by ``src/core/i18n.js``.
...
+-   "core MODULENAME" for changes to core modules in src/core.
```
Comment (stevepiercy) [inline README.md:285]:
```suggestion
-   "core MODULENAME" for changes to core modules in `src/core`.
```

File: README.md [hunk 6]
```diff
@@ -232,3 +252,78 @@ The translations are handled by ``src/core/i18n.js``.
...
+
+A commit message with a changelog-grouping must be structured like so: `TYPE(GROUP): MESSAGE`.
```
Comment (stevepiercy) [inline README.md:287]:
```suggestion
A commit message with a changelog grouping must be structured like so: `TYPE(GROUP): MESSAGE`.
```

File: README.md [hunk 7]
```diff
@@ -232,3 +252,78 @@ The translations are handled by ``src/core/i18n.js``.
...
+Without grouping: `TYPE: MESSAGE`
+
+If the commit message doesn't follow this convention it won't be included in the changelog.
```
Comment (stevepiercy) [inline README.md:290]:
```suggestion
If the commit message doesn't follow this convention, then it won't be included in the changelog.
```

File: README.md [hunk 8]
```diff
@@ -232,3 +252,78 @@ The translations are handled by ``src/core/i18n.js``.
...
+To bypass the pre-commit hook, use the git `-n` switch.
+Example: `git commit yarn.lock -m"yarn install." -n`.
```
Comment (stevepiercy) [inline README.md:292]:
```suggestion
Example: `git commit yarn.lock -m "yarn install." -n`.
```

File: README.md [hunk 9]
```diff
@@ -232,3 +252,78 @@ The translations are handled by ``src/core/i18n.js``.
...
+
+If you are working on a component like the structure pattern (pat-structure), use `pat structure` as a group.
+
+Examples:
+
+Add a feature to the structure pattern:
+
+```
+git commit src/pat/structure -m"feat(pat structure): Add feature to cook some coffee"
```
Comment (stevepiercy) [inline README.md:301]:
```suggestion
git commit src/pat/structure -m "feat(pat structure): Add feature to cook some coffee"
```

File: README.md [hunk 10]
```diff
@@ -232,3 +252,78 @@ The translations are handled by ``src/core/i18n.js``.
...
+```
+
+Cleanup task:
+
+```
+git commit -am"maint(Cleanup): Remove whitespace from documentation."
```
Comment (stevepiercy) [inline README.md:307]:
```suggestion
git commit -am "maint(Cleanup): Remove whitespace from documentation."
```

File: README.md [hunk 11]
```diff
@@ -232,3 +252,78 @@ The translations are handled by ``src/core/i18n.js``.
...
+
+```
+
+or without a grouping:
+
+```
+git commit -am"maint: Remove unnecessary file from root directory."
```
Comment (stevepiercy) [inline README.md:314]:
```suggestion
git commit -am "maint: Remove unnecessary file from root directory."
```

File: README.md [hunk 12]
```diff
@@ -232,3 +252,78 @@ The translations are handled by ``src/core/i18n.js``.
...
+To bypass the pre-commit hook, use the git `-n` switch.
+Example: `git commit yarn.lock -m"yarn install." -n`.
+
+If you are working on a component like the structure pattern (pat-structure), use `pat structure` as a group.
+
+Examples:
+
+Add a feature to the structure pattern:
+
+```
```
Comment (stevepiercy) [inline README.md:300]:
````suggestion
```shell
````

File: README.md [hunk 13]
```diff
@@ -232,3 +252,78 @@ The translations are handled by ``src/core/i18n.js``.
...
+git commit src/pat/structure -m"feat(pat structure): Add feature to cook some coffee"
+```
+
+Cleanup task:
+
+```
```
Comment (stevepiercy) [inline README.md:306]:
````suggestion
```shell
````

File: README.md [hunk 14]
```diff
@@ -232,3 +252,78 @@ The translations are handled by ``src/core/i18n.js``.
...
+git commit -am"maint(Cleanup): Remove whitespace from documentation."
+
+```
+
+or without a grouping:
+
+```
```
Comment (stevepiercy) [inline README.md:313]:
````suggestion
```shell
````

File: README.md [hunk 15]
```diff
@@ -232,3 +252,78 @@ The translations are handled by ``src/core/i18n.js``.
...
+git commit -am"maint: Remove unnecessary file from root directory."
+```
+
+---
+
+**Note**
+
+Please keep commits on the `yarn.lock` file or any other auto-generated code seperate.
+
+Just commit them seperately with `git commit yarn.lock -m"yarn install" -n`.
```
Comment (stevepiercy) [inline README.md:323]:
```suggestion
Just commit them seperately with `git commit yarn.lock -m "yarn install" -n`.
```

File: README.md [hunk 16]
```diff
@@ -232,3 +252,78 @@ The translations are handled by ``src/core/i18n.js``.
...
+
+Having them seperately reduces the effort when merging or rebasing branches where a merge conflict can easily happen.
```
Comment (stevepiercy) [inline README.md:325]:
```suggestion
Keeping them seperate reduces the effort when merging or rebasing branches where a merge conflict can easily happen.
```

File: README.md [hunk 17]
```diff
@@ -232,3 +252,78 @@ The translations are handled by ``src/core/i18n.js``.
...
+In such cases you can just withdraw your changes on the yarn.lock file or remove those commits and re-install with `yarn install` at the end of a successful merge or rebase.
```
Comment (stevepiercy) [inline README.md:326]:
```suggestion
In such cases you can just withdraw your changes on the `yarn.lock` file, or remove those commits and reinstall with `yarn install` at the end of a successful merge or rebase.
```

File: src/pat/preventdoublesubmit/README.md [hunk 18]
```diff
@@ -5,6 +5,8 @@ title: Prevent double submit
 
 # Prevent double submit pattern.
 
+Prevent multiple submissions of the same forn.
```
Comment (stevepiercy) [inline src/pat/preventdoublesubmit/README.md:8]:
```suggestion
Prevent multiple submissions of the same form.
```

File: src/pat/select2/README.md [hunk 19]
```diff
@@ -5,7 +5,8 @@ title: Select2
 
 # Select2 pattern.
 
-Autoselection widget.
+Show a widget which enhances dropdown selections with automatic suggestions, search and tagging functionality.
```
Comment (stevepiercy) [inline src/pat/select2/README.md:8]:
```suggestion
Show a widget which enhances dropdown selections with automatic suggestions, search, and tagging functionality.
```

File: src/pat/textareamimetypeselector/README.md [hunk 20]
```diff
@@ -5,7 +5,10 @@ title: Textarea Mimetype Selector
 
 # Textarea Mimetype Selector pattern.
 
-This pattern displays a mimetype selection widget for textareas. It switches the widget according to the selected mimetype.
+Display the mime type selection widget for textareas.
```
Comment (stevepiercy) [inline src/pat/textareamimetypeselector/README.md:8]:
```suggestion
Display the MIME type selection widget for textareas.
```

File: src/pat/textareamimetypeselector/README.md [hunk 21]
```diff
@@ -5,7 +5,10 @@ title: Textarea Mimetype Selector
...
+
+This pattern displays a mimetype selection widget for textareas.
```
Comment (stevepiercy) [inline src/pat/textareamimetypeselector/README.md:10]:
```suggestion
This pattern displays a MIME type selection widget for textareas.
```

File: src/pat/textareamimetypeselector/README.md [hunk 22]
```diff
@@ -5,7 +5,10 @@ title: Textarea Mimetype Selector
...
+It switches the widget according to the selected mimetype.
```
Comment (stevepiercy) [inline src/pat/textareamimetypeselector/README.md:11]:
```suggestion
It switches the widget according to the selected MIME type.
```

File: src/pat/texteditor/README.md [hunk 23]
```diff
@@ -5,6 +5,15 @@ title: Text Editor
 
 # Text Editor pattern.
 
+Show a code editor.
+
+---
+**Deprecated**
+
+This pattern is deprecated. Use Patternslib' pat-code-editor instead.
```
Comment (stevepiercy) [inline src/pat/texteditor/README.md:13]:
```suggestion
This pattern is deprecated. Use Patternslib's pat-code-editor instead.
```

File: src/pat/tinymce/README.md [hunk 24]
```diff
@@ -3,7 +3,9 @@ permalink: "pat/tinymce/"
 title: tinyMCE
 ---
 
-# tinyMCE pattern.
+# TinyMCE pattern.
```
Comment (stevepiercy) [inline src/pat/tinymce/README.md:6]:
```suggestion
# TinyMCE pattern
```

File: src/pat/tree/README.md [hunk 25]
```diff
@@ -5,6 +5,8 @@ title: Tree
 
 # Tree
 
+Renders a navigatable tree from a data structure.
```
Comment (stevepiercy) [inline src/pat/tree/README.md:8]:
```suggestion
Renders a navigable tree from a data structure.
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]
This is great. I made some minor grammar and MyST syntax changes.

**Review by petschki — APPROVED** [review]
LGTM!

File: README.md [hunk 26]
```diff
@@ -93,7 +39,81 @@ In such cases you can just withdraw your changes on the yarn.lock file or remove
...
 
 
-# Development
+## Mockup overview
+
+Mockup is a JavaScrip UI library which provides widgets, apps and functionality for the Plone Classic UI variant.
+It is based on Patternslib(XXX) and provides it's functionality through so-called Patterns which are initialized and get active if a triggering CSS selector is found in the DOM tree.
+For example, the related items widget is initialized on a form input field with the CSS class `pat-relateditems`.
+The configuration is done via data attributes, in case of the related items pattern it's an attribute called `data-pat-relateditems`.
+The data strucutre of the configuration can be a JSON string or CSS declaration like key-value pairs separated by a semicolon.
```
[the remark this answers is not in the document]
Comment (petschki) [reply README.md:50]: Hm ... this needs maybe more clarification. The key/values are separated with colons but the k/v pairs are separated with semicolon:

```
key: value;
key: value;
...
```

[the remark this answers is not in the document]
Comment (stevepiercy) [reply README.md:50]: I updated my suggestion accordingly. Please let me know. Thank you!

**Comment by petschki** [comment]
I'd say lets publish this state to the documentation. We can fix the gaps afterwards.

**Review by stevepiercy — CHANGES_REQUESTED** [review]
There are two unresolved questions. Once those are answered and fixed, then I approve and we can merge this. Please let me know. Thank you! @petschki @thet

File: README.md [hunk 27]
```diff
@@ -232,3 +160,78 @@ The translations are handled by ``src/core/i18n.js``.
...
+This is enforced via a pre-commit hook managed by [husky](https://github.com/typicode/husky).
+
+**And this is how you use it:**
+
+We have 4 different types of changelog entries:
+
+-   Breaking Changes (type: `breaking`),
+-   Features (type: `feat`),
+-   Bug Fixes (type: `fix`),
+-   Maintenance (type: `maint`).
```
Comment (stevepiercy) [inline README.md:277]: Other repos also have:

```
-   Documentation (type: `documentation`)
```

`.doc` opens in Word and other office apps by default, and this file extension is explicit.

Is that something you want to add?

File: README.md [hunk 28]
```diff
@@ -7,18 +7,18 @@ Mockup is the JavaScript stack of the Plone Classic UI.
 
 There are several options to integrate Mockup.
 
-- First, it comes pre-installed with Plone 6 Classic UI.
-- Second, you can install and compile a bundle. See below.
-- Third you can download a ZIP from Mockup's release pages on GitHub.
-- Fourth, you can use a precompiled bundle from a CDN, like:
+1.  It comes pre-installed with Plone 6 Classic UI.
+2.  You can install and compile a bundle. See [insert anchor to specific section, as "below" is vague].
```
[the remark this answers is not in the document]
Comment (petschki) [reply README.md]: I've linked that to the `Development` section.

File: README.md [hunk 29]
```diff
@@ -41,10 +41,12 @@ There are several options to integrate Mockup.
 
 ## Mockup overview
 
-Mockup is a JavaScrip UI library which provides widgets, apps and functionality for the Plone Classic UI variant.
-It is based on Patternslib(XXX) and provides it's functionality through so-called Patterns which are initialized and get active if a triggering CSS selector is found in the DOM tree.
+Mockup is a JavaScript UI library which provides widgets, apps, and functionality for the Plone Classic UI frontend.
+It is based on Patternslib(XXX), and provides its functionality through patterns.
```
[the remark this answers is not in the document]
Comment (petschki) [reply README.md:45]: This should show the current used Patternslib version from `package.json`. Is it possible to extract that in Markdown automatically?

[the remark this answers is not in the document]
Comment (stevepiercy) [reply README.md:45]: We currently manually update the version of packages in our `conf.py`.

https://github.com/plone/documentation/blob/6.0/docs/conf.py#L306-L327

The dict could be extended automatically. Then in the MyST file:

https://github.com/plone/documentation/blob/e011a6df4e8444a2f7ae59ec9678aa7ed60bf374/docs/install/manage-add-ons-packages.md?plain=1#L64

File: README.md [hunk 30]
```diff
@@ -232,3 +254,78 @@ The translations are handled by ``src/core/i18n.js``.
...
+
+```shell
+git commit src/pat/structure -m "feat(pat structure): Add feature to cook some coffee"
+```
+
+Cleanup task:
+
+```shell
+git commit -am "maint(Cleanup): Remove whitespace from documentation."
+
```
Comment (stevepiercy) [inline README.md:310]:
```suggestion
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]

File: README.md (hunk 27 above)
Comment (petschki) [reply README.md:277]: We could add it in mokcup but we currently reuse `release-it` config from [@patternslib/dev](https://github.com/Patternslib/dev/blob/main/.release-it.js) so it should be extended there first before we mention it here.

@thet what do you think?

Comment (MrTango) [reply README.md:277]: maybe open an issue for this feature and merge this PR

File: README.md (hunk 29 above)
[the remark this answers is not in the document]
Comment (stevepiercy) [reply README.md]: Done in [5f5f92f](https://github.com/plone/mockup/pull/1345/commits/5f5f92f5dd9168540356ee0eb92874400d23c0b5)

File: README.md (hunk 27 above)
Comment (stevepiercy) [reply README.md:277]: Issue created: https://github.com/plone/mockup/issues/1373

**Review by stevepiercy — APPROVED** [review]

**Comment by stevepiercy** [comment]
I will merge if it goes green. Thank you!

**Pull request merged**